src/Collection.php

```diff
@@ -156,40 +155,32 @@
 final class Collection implements CollectionInterface
 {
     /**
-     * @var iterable<int, mixed>
+     * @var IteratorAggregate<TKey, T>
      */
-    private iterable $parameters;
+    private IteratorAggregate $innerIterator;
 
     /**
-     * @var callable(mixed ...$parameters): iterable<TKey, T>
-     */
-    private $source;
-
-    /**
-     * @psalm-external-mutation-free
```

Comment:
is removing this intentional?

Reply:
Yeah, I think we should even get rid of them everywhere. psalm doesn't handle them correctly anyway. I really wanted to avoid updating the baselines too much.